### ISSUE TITLE
fstatat: socket is not connected

### ISSUE THREAD

**Issue by firefoxlover** [fish-shell/fish-shell#9674]
When I use fish, after entering 3 letters like sss or anything else, I get 

```
user@fedora /v/h/user> afstatat: The socket is not connected
fstatat: The socket is not connected
```

Using bash I dont get this error.

I dont really know what I did. I looked all my directories in PATH and they are normal and existent. I renamed the fish conf so vanilla doesnt solve it either. I tried Konsole and XTerm.

I can enter commands like bash, ls, cd, but not things like "update".

fish version 3.6.0

Operating System: Fedora Linux 37
Kernel Version: 6.1.18-200.fc37.x86_64 (64-bit)
Terminal: Konsole (KDE) and XTerm

**Comment by faho** [comment]
Duplicate of #9550

**Comment by faho** [comment]
>I dont really know what I did. I looked all my directories in PATH and they are normal and existent. I renamed the fish conf so vanilla doesnt solve it either. I tried Konsole and XTerm.

You want to look at $CDPATH.

**Comment by firefoxlover** [comment]
I already saw that issue.

```
echo $CDPATH
```
returns nothing on bash or fish.

**Comment by faho** [comment]
This error is garbage - it gives an "ENOTCONN", which is not an error that fstatat is supposed to return, or that really makes any sense.

This is a case where I would check that your system doesn't have a bug - like a broken filesystem somewhere.

Check if any directory in your $PATH is on some awkward filesystem - we've seen weird errors on e.g. mounted google drive things.

**Comment by faho** [comment]
This isn't really a duplicate of  #9550 - same *idea* (your system is haunted and returns broken errors), but not the same fix.

I'm gonna reopen this, and then we're gonna remove the error printing here because there's nothing actionable about the error (which is only printed in a wildcard expansion context).

**Comment by firefoxlover** [comment]
reboot solved it somehow

**Comment by faho** [comment]
(@zanchey - cheeky fix for 3.6.1, don't think this deserves a changelog)

**Comment by sam-mccall** [comment]
@faho thanks for making this blanket fix!

FWIW I was seeing ENOKEY too, and it was sufficient to have a symlink pointing onto the network filesystem in my working directory (without me ever trying to access it). Presumably because the relevant call to fstatat was not passing `AT_SYMLINK_NOFOLLOW`.

(I understand the argument that FSes shouldn't return errors outside spec, but as a user who needs to access these things I can't do much about this other than not use fish.)
